lib/msal-browser/src/app/PublicClientApplication.ts

```diff
@@ -3,6 +3,190 @@
+     * If your application supports Accounts in any organizational directory, replace "Enter_the_Tenant_Info_Here" value with organizations.
+     * If your application supports Accounts in any organizational directory and personal Microsoft accounts, replace "Enter_the_Tenant_Info_Here" value with common.
+     * To restrict support to Personal Microsoft accounts only, replace "Enter_the_Tenant_Info_Here" value with consumers.
+     *
+     * In Azure B2C, authority is of the form https://&lt;instance&gt;/tfp/&lt;tenant&gt;/&lt;policyName&gt;/
+     *
+     * @param {@link (Configuration:type)} configuration object for the MSAL PublicClientApplication instance
+     */
+    constructor(configuration: Configuration) {
+        // Set the configuration
+        this.config = buildConfiguration(configuration);
+
+        this.authModule = new msalAuth.AuthorizationCodeModule({
+            auth: this.config.auth,
+            cryptoInterface: null,
+            networkInterface: null,
+            storageInterface: null
+        });
+    }
+
+    // #region Redirect Flow
+
+    /**
+     * Set the callback functions for the redirect flow to send back the success or error object.
+     * @param {@link (authCallback:type)} authCallback - Callback which contains
+     * an AuthError object, containing error data from either the server
+     * or the library, depending on the origin of the error, or the AuthResponse object 
+     * containing data from the server (returned with a null or non-blocking error).
+     */
+    handleRedirectCallback(authCallback: authCallback): void {
+        throw new Error("Method not implemented.");
+    }
+
+    /**
+     * Use when initiating the login process by redirecting the user's browser to the authorization endpoint.
+     * @param {@link (AuthenticationParameters:type)}
+     */
+    loginRedirect(request: msalAuth.AuthenticationParameters): void {
+        throw new Error("Method not implemented.");
+    }
+
+    /**
+     * Use when you want to obtain an access_token for your API by redirecting the user's browser window to the authorization endpoint.
+     * @param {@link (AuthenticationParameters:type)}
+     *
+     * To renew idToken, please pass clientId as the only scope in the Authentication Parameters
+     */
+    acquireTokenRedirect(request: msalAuth.AuthenticationParameters): void {
+        throw new Error("Method not implemented.");
+    }
+
+    // #endregion
+
+    // #region Popup Flow 
+
+    /**
+     * Use when initiating the login process via opening a popup window in the user's browser
+     *
+     * @param {@link (AuthenticationParameters:type)}
+     *
+     * @returns {Promise.<TokenResponse>} - a promise that is fulfilled when this function has completed, or rejected if an error was raised. Returns the {@link AuthResponse} object
+     */
+    loginPopup(request: msalAuth.AuthenticationParameters): Promise<msalAuth.TokenResponse> {
```

Comment:
this may be more for my own learning, but can you clarify the difference in functionality between `loginPopup` and `loginRedirect`? Is the latter meant to be silent?